### PR TITLE
Fix/user flow bugs

### LINKED ISSUE #84: BUG: User Flow - trip timer expiration
- [x] BUG FIXED

Two buggy user flows have occurred, so far:
1.  Trip timer expires and instantly restarts at 60 seconds, no modal / messaging (only one occurrence, no cause determined).
2.  Trip timer expires and "AddTime" window appears, and user selects a time extension.  Timer restarts with correct amount of extension time.  That timer expires, and timer simply display zeros.  No reroute, modal popup, etc.  Need to get it to either reroute to "Plan Trip" view or have it trigger the "AddTime" window, again.

Video of second user flow:

https://user-images.githubusercontent.com/51062974/126381784-4065120a-d25d-42fe-9e9b-b0d0f96d3b0f.mov

### LINKED ISSUE #86: BUG: User Flow - ETA window stuck on "loading" state
- [x] BUG FIXED

- ETA window gets stuck on "loading" display when estimated time is < 1 minute (value rounds down to zero, so conditional rendering of ETA message fails).  
- Replicate this User Flow by selecting two addresses on the same block

Screen Shot 2021-07-20 at 11 30 14 AM

### PR DESCRIPTION
**Pull request by pcmueller**
## Type of Change
- [x] JSX
- [ ] HTML
- [x] JavaScript
- [ ] CSS
- [x] React
- [x] React Router
- [ ] Cypress
- [ ] Apollo
- [ ] GraphQL
- [ ] Twilio API
- [x] Refactor/Fix Bugs

## Please describe the change made
User Stories bug fixes

## Please give a short description of the changes that were made below

Fix:
- Refactored conditional rendering / Hooks in multiple components to prevent user flow from getting stuck on "loading" display
- Refactored conditional rendering / Hooks in multiple components to prevent trip timer from expiring and getting stuck on "empty timer" view - now when the timer expires on Current Trip view we automatically render the AddTime modal, Alert modal, or `<Redirect/>` to the Main View if neither modal is relevant.
- Fixed the 'trip started' SMS message so that it displays either 'minutes' or 'seconds' after the ETA time, whichever is the appropriate unit.
- Eliminated most dev tools console warnings: the only ones left are related to unused query/mutation variables.  I didn't wanna start changing anything there, since I'm less familiar with the code.  I've attached a screenshot of those remaining warnings.

Feature:
- Built out functionality so that if the ETA is < 1 minute (if starting and ending address are both on the same block, for example), we default to a 30 second ETA / timer.

## How has this been tested?
- [x] In Browser
- [x] In Console
- [ ] Cypress
- [ ] Postman

Screen Shot 2021-07-20 at 12 36 54 PM

### CODE REVIEW

File: src/Components/AddContact/AddContact.js [hunk 1]
```diff
@@ -42,18 +42,18 @@ function AddContact({ user }) {
...
-    if(countryCode.length === 0){
-      setPhoneVerify(true)
-      return true
-    } else if(areaCode.length !== 3){
-      setPhoneVerify(true)
-      return true
-    } else if(phoneNumber.length !== 7){
-      setPhoneVerify(true)
-      return true
+    if (countryCode.length === 0) {
```
Comment (carolineshowscode) [inline src/Components/AddContact/AddContact.js:45]: thank you for correcting my spacing

File: src/Components/CurrentTrip/CurrentTrip.js [hunk 2]
```diff
@@ -33,55 +33,76 @@ function CurrentTrip({ user, eta, contact }) {
...
+      setEtaSeconds(null);
+      setExtension(0);
+      setTripEnded(true);
+      TripNotCompleteMessage(user, contact)
     }
-  }, [emergency])
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [emergency]);
+
+  function beginTrip() {
```
Comment (carolineshowscode) [inline src/Components/CurrentTrip/CurrentTrip.js:87]: nice job working on this logic!

**Review by carolineshowscode — APPROVED** [review]
great work!

File: src/Components/AboutUs/AboutUs.js [hunk 3]
```diff
@@ -34,11 +34,16 @@ function AboutUs({ user }) {
...
       }
     });
     const beCards = devNames.map(dev => {
-      if (devNames.indexOf(dev) > 2)
-      return buildDevCard(dev);
+      if (devNames.indexOf(dev) > 2) {
+        return buildDevCard(dev);
+      } else {
+        return undefined;
+      }
```
Comment (bhohnco) [inline src/Components/AboutUs/AboutUs.js:46]: Making us look good, nice!

File: src/Components/CurrentTrip/CurrentTrip.js [hunk 4]
```diff
@@ -33,55 +33,76 @@ function CurrentTrip({ user, eta, contact }) {
...
-    // TripCompleteMessage(user,contact)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [tripEnded]);
 
   useEffect(() => {
-    if (!tripIsActive && !tripEnded) {
-      setEtaModalIsOpen(true);
+    if (extension.value > 0) {
+      setEtaSeconds(extension.value);
+      TripExtendedMessage(user, extension, contact)
```
Comment (bhohnco) [inline src/Components/CurrentTrip/CurrentTrip.js:70]: Solid stuff Pete!

**Review by gaelyn — APPROVED** [review]
Looks fantastic team. Do we need to do any phone number formatting validations on the backend? 🤔

**Review by joemecha — APPROVED** [review]
Looks like some solid bug zapping here. Great to see the passing build in Travis too.

**Pull request merged**